types/src/vm_error.rs

```diff
@@ -469,6 +469,7 @@ pub enum StatusCode {
     CODE_DESERIALIZATION_ERROR = 4019,
     EXECUTION_STACK_OVERFLOW = 4020,
     CALL_STACK_OVERFLOW = 4021,
+    NATIVE_FUNCTION_ERROR = 4022,
```

Comment:
What are we currently doing for errors in (e.g.) native crypto functions?

Reply:
That was exactly what I did. Previously the native crypto functions are treated as assertion failures. The returned status code will be `ABORTED`. It seems to me that it is not ideal as we will conflate the errors code namespace between assertions failures and native function failures cc @tzakian